File: src/main/com/dune/DuneLanguage.java
package com.dune;

import com.intellij.lang.Language;

public class DuneLanguage extends Language {
    public static final DuneLanguage INSTANCE = new DuneLanguage();
    public static final String NAME = "Dune";

    private DuneLanguage() {
        super(NAME.toUpperCase());
    }
}

File: src/main/kotlin/com/dune/language/DuneLanguage.kt
package com.dune.language

import com.intellij.lang.Language

/**
 * Dune Language
 */
object DuneLanguage : Language("Dune") {
    private fun readResolve(): Any = DuneLanguage
    override fun getDisplayName(): String = "Dune"
    override fun isCaseSensitive(): Boolean = true
}